### PR TITLE
Add filter on fluorescence experimental errors

### LINKED ISSUE #1234: When IC50 stderrors are very large pIC50 and IC50 values can be clipped to very large numbers.
Using stderr based rounding for IC50s and pIC50s here: https://github.com/asapdiscovery/asapdiscovery/blob/main/asapdiscovery-data/asapdiscovery/data/util/utils.py#L951-L956

Can make values very large if stderrs are large. We should work on some kind of fix.

### PR DESCRIPTION
**Pull request by kaminow**
Add filter on fluorescence experimental errors to prevent overly large errors from rounding the readout values to 0/10. Closes #1234.

### CODE REVIEW

**Comment by kaminow** [comment]
@hmacdope I think this is ready, lmk if you think there's anything else we want to do. rather than removing stuff here I just set the IC50 and pIC50 values to NaNs since they should be removed later in the pipeline anyway, but we can change if you think we should just remove the offending rows

File: asapdiscovery-data/asapdiscovery/data/util/utils.py [hunk 1]
```diff
@@ -952,21 +956,28 @@ def parse_fluorescence_data_cdd(
...
-                    pIC50, uncertainty=pIC50_stderr, sep=tuple, output_type=str
-                )  # strings
-            except ModuleNotFoundError:
-                # Don't round
-                pass
+            # Check error for filtering
+            if pIC50_stderr > pic50_stderr_filt:
+                IC50 = "nan"
+                pIC50 = "nan"
+                # @HMO do we want to leave the stderr values as they are for posterity
```
Comment (hmacdope) [inline asapdiscovery-data/asapdiscovery/data/util/utils.py:963]: Log them as a warning I think would be good.

File: asapdiscovery-data/asapdiscovery/data/util/utils.py [hunk 2]
```diff
@@ -952,21 +956,28 @@ def parse_fluorescence_data_cdd(
...
+                #  or set them to NaN as well?
```
Comment (hmacdope) [inline asapdiscovery-data/asapdiscovery/data/util/utils.py:964]: Set them to `nan` I think is the right move.

File: asapdiscovery-ml/asapdiscovery/ml/cli_mlops.py [hunk 3]
```diff
@@ -774,13 +781,20 @@ def mlops():
     is_flag=True,
     help="Run in test mode, no S3 push or WandB logging to main project",
 )
+@click.option(
+    "--pic50-stderr-filt",
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/cli_mlops.py:785]: Nice, thanks.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Great work! I say we leave in the backup clipping L488-498, what do you think?

**Comment by hmacdope** [comment]
Also @kaminow you will need to merge with `upstream/main`

**Comment by kaminow** [comment]
@hmacdope I think ready for another review!

also, which file are you referring to here?
> Great work! I say we leave in the backup clipping L488-498, what do you think?

**Comment by hmacdope** [comment]
> @hmacdope I think ready for another review!
> 
> also, which file are you referring to here?
> 
> > Great work! I say we leave in the backup clipping L488-498, what do you think?

asapdiscovery-ml/asapdiscovery/ml/cli_mlops.py , but should be all good.

**Review by hmacdope — APPROVED** [review]

**Pull request merged**